app/controllers/incomeSources/manage/ManageIncomeSourceDetailsController.scala

```diff
@@ -153,11 +156,13 @@ class ManageIncomeSourceDetailsController @Inject()(val view: ManageIncomeSource
                 incomeSourceType = SelfEmployment
               )
 
-              result.flatMap(
-                sessionService.set(SessionKeys.incomeSourceId, id, _)
-              ).flatMap {
-                case Right(result) => Future.successful(result)
-                case Left(exception) => Future.failed(exception)
+              sessionService.createSession(JourneyType(Manage, SelfEmployment).toString).flatMap {
+                case true =>
```

Comment:
* createSession would alway return value of true, so checking if result true or false is redundant (maybe we need to return Unit from .createSession ?);
* I am not sure about this, but I think that we need to createSession first before handle actual request, as the outcome of creating session can be fail; but this something we need to agreed across all journeys (if we should apply any order at all ?)

Reply:
To create a session you need to know the operation (Add/Manage/Cease) and also the incomeSourceType(Self Employment/Uk/Foreign), this is the earliest point in the journey where you can know that the user has chosen a Self Employment business

Reply:
I was actually referring to next change, which working fine as ATs are passing;
there is not big different from how is works right now, but I really think that hitting Mongo first and get result 
 would be more appropriate before processing request

```
              sessionService.createSession(JourneyType(Manage, SelfEmployment).toString).flatMap {
                case true =>
                  sessionService.setMongoKey(ManageIncomeSourceData.incomeSourceIdField, id, JourneyType(Manage, SelfEmployment)).flatMap {
                    case Right(_) =>
                      handleRequest(
                        sources = mtdItUser.incomeSources,
                        isAgent = true,
                        backUrl = controllers.incomeSources.manage.routes.ManageIncomeSourceController.show(true).url,
                        id = Some(id),
                        incomeSourceType = SelfEmployment
                      )
                    case Left(exception) => Future.failed(exception)
                  }
                case false => Future.failed(new Error("Failed to create mongo session"))
              }
```